databox/api/src/Asset/Attribute/TemplateResolver.php

```diff
@@ -17,10 +17,13 @@ public function __construct()
         ]);
     }
 
-    public function resolve(string $template, array $values): string
+    public function resolve($template, array $values): string
```

Comment:
Please keep strong type hint here as the type check should be outside.

Reply:
the template can be a number

Reply:
If so, this resolver should not be called

Reply:
only the resolver can decide if the template is twig code, so it's already too late.
if we want to **not** call the resolver for a number/bool..., then we must move the "{" test before the call, thus revealing outside the resolver that it internally uses twig syntax.
Or we must implement some kind of resolver->isTemplate() ?

Reply:
> only the resolver can decide if the template is twig code

Why? We can have a `str_contains($template, '{')` in this service (which only accepts string, remember this service is already used by other services), and conditionally call this resolver only if the input is a string.

So you can check if `$o` is a string in `ModuleOptionsResolver`:

```php
$r[$k] = is_string($o) ? $this->templateResolver->resolve($o, $context) : $o;
```

This is principle of responsibilities, ModuleOptionsResolver can deal with any type of values, but TemplateResolver only resolve template strings.